src/main/java/org/sagebionetworks/bridge/exporter3/Exporter3WorkerProcessor.java

```diff
@@ -203,26 +186,64 @@ void process(Exporter3Request request) throws BridgeSynapseException, IOExceptio
             return;
         }
 
+        // Which study is this export for?
+        Set<String> studyIdSet;
+        Set<String> participantStudyIdSet = new HashSet<>(participant.getStudyIds());
+        Map<String, String> metadataMap = makeMetadataFromRecord(record);
+        String scheduleGuid = metadataMap.get(METADATA_KEY_SCHEDULE_GUID);
+        if (scheduleGuid != null) {
+            // If the timeline metadata contains a schedule Guid, filter the participant's studies by the studies that
+            // this health data is a part of.
+            List<String> timelineStudyIdList = bridgeHelper.getStudyIdsUsingSchedule(appId, scheduleGuid);
+            Set<String> timelineStudyIdSet = new HashSet<>(timelineStudyIdList);
+            studyIdSet = Sets.intersection(participantStudyIdSet, timelineStudyIdSet);
+        } else {
+            // If there is no timeline metadata, just export over all of the participant's studies.
+            studyIdSet = participantStudyIdSet;
+        }
+
+        // Filter sets even futher, based on which ones are configured.
+        List<Study> studiesToExport = new ArrayList<>();
+        for (String studyId : studyIdSet) {
+            Study study = bridgeHelper.getStudy(appId, studyId);
+            if (BridgeUtils.isExporter3Configured(study)) {
+                studiesToExport.add(study);
+            }
+        }
+
+        if (!exportForApp && studiesToExport.isEmpty()) {
+            // No destinations to export to. Skip.
+            return;
+        }
+
         // Copy the file to the raw health data bucket. This includes folderization.
         // Note that in Exporter 3.0, upload ID is the same as record ID.
         Upload upload = bridgeHelper.getUploadByUploadId(recordId);
-        ExportContext exportContext;
+        String hexMd5;
         if (upload.isEncrypted()) {
-            exportContext = decryptAndUploadFile(appId, upload, record);
+            hexMd5 = decryptAndUploadFile(app, upload, record, metadataMap, exportForApp, studiesToExport);
         } else {
-            exportContext = copyUploadToHealthDataBucket(appId, upload, record);
+            hexMd5 = copyUploadToHealthDataBucket(app, upload, record, metadataMap, exportForApp, studiesToExport);
         }
 
         // Upload to Synapse.
-        exportToSynapse(appId, exporter3Config, upload, record, exportContext);
+        if (exportForApp) {
+            exportToSynapse(appId, null, app.getExporter3Configuration(), upload, record, metadataMap, hexMd5);
+        }
+        for (Study study : studiesToExport) {
+            exportToSynapse(appId, study.getIdentifier(), study.getExporter3Configuration(), upload, record,
+                    metadataMap, hexMd5);
+        }
```

Comment:
Add integ test so that if we export to study A and we export to study A again (eg a redrive, or one study failed and the task is vended by SQS again).